src/snowflake/snowpark/__init__.py

```diff
@@ -69,3 +72,22 @@
     WhenNotMatchedClause,
 )
 from snowflake.snowpark.window import Window, WindowSpec
+
+_deprecation_warning_msg = (
+    "Snowpark Python Runtime Support for Python 3.8 will reach end-of-life "
+    "on March 31, 2025. To ensure continued support, please upgrade "
+    "your runtime environment to Python > 3.8 before this date. "
+    "For more details, please refer "
+    "to https://docs.snowflake.com/en/developer-guide/python-runtime-support-policy."
```

Comment:
@sfc-gh-deliu , @sfc-gh-qding Please review this text.

Reply:
"Python > 3.8" seems kind of awkward. How about something like "upgrade your runtime environment to a newer version before this date."

Also, are there any versions greater than 3.8 that we don't yet fully support?

Reply:
@sfc-gh-pspaeth thanks for the feedback!

> Also, are there any versions greater than 3.8 that we don't yet fully support?
we currently support 3.8, 3.9, 3.10, 3.11.
3.12 and 3.13 are not supported yet

Reply:
So we can't say greater than 3.8, we have to specify versions 3.9 - 3.11.

Reply:
updated